packages/protocol/script/deploy.s.sol

```diff
@@ -0,0 +1,240 @@
+// SPDX-License-Identifier: MIT
+//  _____     _ _         _         _
+// |_   _|_ _(_) |_____  | |   __ _| |__ ___
+//   | |/ _` | | / / _ \ | |__/ _` | '_ (_-<
+//   |_|\__,_|_|_\_\___/ |____\__,_|_.__/__/
+
+pragma solidity ^0.8.18;
+
+import "forge-std/Script.sol";
+import "forge-std/console2.sol";
+import "@openzeppelin/contracts/proxy/transparent/TransparentUpgradeableProxy.sol";
+import "../contracts/L1/TaikoToken.sol";
+import "../contracts/L1/TaikoL1.sol";
+import "../contracts/bridge/Bridge.sol";
+import "../contracts/bridge/TokenVault.sol";
+import "../contracts/signal/SignalService.sol";
+import "../contracts/thirdparty/AddressManager.sol";
+import "../contracts/test/erc20/FreeMintERC20.sol";
+import "../contracts/test/erc20/MayFailFreeMintERC20.sol";
+
+contract DeployOnL1 is Script {
```

Comment:
just a comment: maybe this script pattern: https://twitter.com/PaulRBerg/status/1629480999888273412?s=20 could be useful for future scripts so they don't need to do any setup.